src/pages/index.js

```diff
@@ -41,70 +55,243 @@ addFormValidator.enableValidation();
+    //   addInitialCards(initialCards)
+    //     .then(() => api.getInitialCards())
+    //     .then((newCards) => {
+    //       console.log("New cards added from initial cards:", newCards);
+    //       section.renderItems(newCards);
+    //     });
+    // } else {
+    //   console.log("Rendering cards from the server:", cards);
+    //   section.renderItems(items);
+    // }
+  })
+  .catch((error) => console.error("Error loading data", error));
+
+const editAvatarModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: (data) => {
+    const userAvatar = {
+      avatar: data.avatar,
+    };
+    api
+      .updateUserAvatar(userAvatar)
+      .then((updatedUserInfo) => {
+        userInfo.setUserInfo(updatedUserInfo);
+        editAvatarModal.close();
+      })
+      .catch((error) => console.error("Error updating user avatar:", error));
+  },
+});
+
+const editProfileModal = new PopupWithForm({
+  popupSelector: "#edit-profile-modal",
+  // handleFormSubmit: (data) => {
+  //   userInfo.setUserInfo({ name: data.name, job: data.job });
+  // },
+  handleFormSubmit: (data) => {
+    const userData = {
+      name: data.name,
+      about: data.about,
+    };
+    api
+      .updateUserInfo(userData)
+      .then((updatedUserInfo) => {
+        userInfo.setUserInfo(updatedUserInfo);
+        editProfileModal.close();
+      })
+      .catch((error) => console.error("Error updating user info:", error));
+  },
+});
+
+const addCardModal = new PopupWithForm({
+  popupSelector: "#add-card-modal",
+  // handleFormSubmit: (data) => {
+  //   const cardElement = createCard({ name: data.title, link: data.url });
+  //   section.addItem(cardElement);
+  //   cardForm.reset();
+  //   toggleAddButtonState();
+  // },
+  handleFormSubmit: (data) => {
+    const cardData = {
+      name: data.title,
+      link: data.url,
+    };
+    api
+      .addCard(cardData)
+      .then((newCard) => {
+        if (newCard && newCard.name && newCard.link) {
+          const cardElement = createCard({
+            name: newCard.name,
+            link: newCard.link,
+            _id: newCard._id,
+          });
+          section.addItem(cardElement);
+          cardForm.reset();
+          toggleAddButtonState();
+          addCardModal.close();
+        } else {
+          console.error("Invalid card data:", newCard);
+        }
+      })
+      .catch((error) => console.error("Error adding new card:", error));
+  },
+});
+
+const deleteCardModal = new PopupWithConfirm({
+  popupSelector: "#delete-card-modal",
+  handleConfirm: () => {
+    if (currentCard) {
+      handleDeleteCard(currentCard);
+      deleteCardModal.close(); // Close modal after handling deletion
+    }
+  },
+});
+
+// function addInitialCards(cards) {
+//   const promises = cards.map((cardData) => {
+//     console.log("Adding card:", cardData);
+//     return api
+//       .addCard(cardData)
```

Comment:
**Could be improved**

Please, remove **commented code** to clean up the project